lib/models/JwkEd25519.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Model for representing a Ed25519 key in a JWK format.
+ */
+export default interface JwkEd25519 {
```

Comment:
I'd think it would be safer to type the private key as a separate type so that an API can confirm that only public data is being passed in here. By adding the `d` as optional rather than as a separate interface and then also building a type union of only the public key interfaces we would be able to leverage the type system to filter private keys which is not possible with this approach.

Reply:
I copied the [JwkEs256K model](https://github.com/decentralized-identity/ion-sdk/blob/61750e8/lib/models/JwkEs256k.ts) which follows the same pattern. I agree with you, but this is out of scope for this change

Reply:
Sounds good I opened #22 to track this in a separate PR